diff --git a/pages/backtest.py b/pages/backtest.py
index 5165f06..0fb05c2 100644
--- a/pages/backtest.py
+++ b/pages/backtest.py
@@ -15,6 +15,8 @@ from strategies.kdj import KDJ, KDJBacktest, KDJInfo
 from strategies.macd import MACD, MACDBacktest, MACDInfo
 from strategies.rsi import RSI, RSIBacktest, RSIInfo
 from strategies.stairs import Stairs, StairsBacktest, StairsInfo
+from strategies.triple_golden_cross import TripleGoldenCross, \
+    TripleGoldenCrossBacktest, TripleGoldenCrossInfo
 from strategies.turtle import Turtle, TurtleBacktest, TurtleInfo
 from strategies.wr import WR, WRBacktest, WRInfo

@@ -238,6 +240,7 @@ class Backtest(QWidget):
         macd_info = MACDInfo()
         rsi_info = RSIInfo()
         stairs_info = StairsInfo()
+        triple_golden_cross_info = TripleGoldenCrossInfo()
         turtle_info = TurtleInfo()
         wr_info = WRInfo()
         if self.current_strategy_name == boll_info.name:
@@ -264,6 +267,11 @@ class Backtest(QWidget):
             self.backtest_thread = StairsBacktest(stocks, s_date, e_date,
                                                   init_money, fee,
                                                   pass_fee, tax)
+        elif self.current_strategy_name == triple_golden_cross_info.name:
+            self.backtest_thread = TripleGoldenCrossBacktest(stocks, s_date,
+                                                             e_date,
+                                                             init_money, fee,
+                                                             pass_fee, tax)
         elif self.current_strategy_name == turtle_info.name:
             self.backtest_thread = TurtleBacktest(stocks, s_date, e_date,
                                                   init_money, fee,
@@ -329,6 +337,7 @@ class Backtest(QWidget):
         macd_info = MACDInfo()
         rsi_info = RSIInfo()
         stairs_info = StairsInfo()
+        triple_golden_cross_info = TripleGoldenCrossInfo()
         turtle_info = TurtleInfo()
         wr_info = WRInfo()
         if self.current_strategy_name == boll_info.name:
@@ -343,6 +352,8 @@ class Backtest(QWidget):
             backtest = RSI()
         elif self.current_strategy_name == stairs_info.name:
             backtest = Stairs()
+        elif self.current_strategy_name == triple_golden_cross_info.name:
+            backtest = TripleGoldenCross()
         elif self.current_strategy_name == turtle_info.name:
             backtest = Turtle()
         elif self.current_strategy_name == wr_info.name:
diff --git a/pages/choose.py b/pages/choose.py
index 3966a91..3e47a7b 100644
--- a/pages/choose.py
+++ b/pages/choose.py
@@ -12,6 +12,8 @@ from strategies.kdj import KDJChoose, KDJInfo
 from strategies.macd import MACDChoose, MACDInfo
 from strategies.rsi import RSIChoose, RSIInfo
 from strategies.stairs import StairsChoose, StairsInfo
+from strategies.triple_golden_cross import TripleGoldenCrossChoose, \
+    TripleGoldenCrossInfo
 from strategies.turtle import TurtleChoose, TurtleInfo
 from strategies.volume_increase import VolumeIncreaseChoose, VolumeIncreaseInfo
 from strategies.wr import WRChoose, WRInfo
@@ -43,6 +45,7 @@ class Choose(Plots):
         self.turtle_info = TurtleInfo()
         self.volume_increase_info = VolumeIncreaseInfo()
         self.wr_info = WRInfo()
+        self.triple_golden_cross_info = TripleGoldenCrossInfo()
         self.boll_info = BOLLInfo()
         self.macd_info = MACDInfo()
         self.kdj_info = KDJInfo()
@@ -265,6 +268,10 @@ class Choose(Plots):
                         self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.wr_info.name:
                     self.choose_thread = WRChoose(self.stocks_to_be_chosen)
+                elif self.apply_strategies[0] == \
+                        self.triple_golden_cross_info.name:
+                    self.choose_thread = TripleGoldenCrossChoose(
+                        self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.boll_info.name:
                     self.choose_thread = BOLLChoose(self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.macd_info.name:
diff --git a/pages/nav.py b/pages/nav.py
index 666d61e..dcc0742 100644
--- a/pages/nav.py
+++ b/pages/nav.py
@@ -16,6 +16,8 @@ from strategies.kdj import KDJConfig, KDJInfo
 from strategies.macd import MACDConfig, MACDInfo
 from strategies.rsi import RSIConfig, RSIInfo
 from strategies.stairs import StairsConfig, StairsInfo
+from strategies.triple_golden_cross import TripleGoldenCrossConfig, \
+    TripleGoldenCrossInfo
 from strategies.turtle import TurtleConfig, TurtleInfo
 from strategies.wr import WRConfig, WRInfo
 from strategies.volume_increase import VolumeIncreaseConfig, VolumeIncreaseInfo
@@ -72,6 +74,7 @@ class Nav(QWidget):
         self.turtle_info = TurtleInfo()
         self.volume_increase_info = VolumeIncreaseInfo()
         self.wr_info = WRInfo()
+        self.triple_golden_cross_info = TripleGoldenCrossInfo()
         self.boll_info = BOLLInfo()
         self.macd_info = MACDInfo()
         self.kdj_info = KDJInfo()
@@ -102,6 +105,11 @@ class Nav(QWidget):
                 'choose': 'Y' if self.wr_info.choose_flag else 'N',
                 'watch': 'Y' if self.wr_info.watch_flag else 'N'
             },
+            {
+                'name': self.triple_golden_cross_info.name,
+                'choose': 'Y' if self.triple_golden_cross_info.choose_flag else 'N',
+                'watch': 'Y' if self.triple_golden_cross_info.watch_flag else 'N'
+            },
             {
                 'name': self.boll_info.name,
                 'choose': 'Y' if self.boll_info.choose_flag else 'N',
@@ -320,6 +328,10 @@ class Nav(QWidget):
             cfg_dlg = WRConfig(self)
             cfg_dlg.show()
             cfg_dlg.exec_()
+        if name == self.triple_golden_cross_info.name:
+            cfg_dlg = TripleGoldenCrossConfig(self)
+            cfg_dlg.show()
+            cfg_dlg.exec_()
         if name == self.boll_info.name:
             cfg_dlg = BOLLConfig(self)
             cfg_dlg.show()
diff --git a/widgets/plots.py b/widgets/plots.py
index 67f6cb5..11df151 100644
--- a/widgets/plots.py
+++ b/widgets/plots.py
@@ -9,6 +9,7 @@ from strategies.dual_line import DualLine
 from strategies.kdj import KDJ
 from strategies.macd import MACD
 from strategies.rsi import RSI
+from strategies.triple_golden_cross import TripleGoldenCross
 from strategies.turtle import Turtle
 from strategies.wr import WR
 from utils.candlestick import CandlestickItem
@@ -463,6 +464,18 @@ class Plots(QWidget):
             k_pen_colors = ['r', 'y']
             vol_data = []
             vol_pen_colors = []
+        elif self.current_indicator_name == self.triple_golden_cross_info.name:
+            triple_golden_cross = TripleGoldenCross()
+            fast_ma = triple_golden_cross.calc_fast_ma(closes)
+            slow_ma = triple_golden_cross.calc_slow_ma(closes)
+            season_ma = triple_golden_cross.calc_season_ma_indicator(
+                closes)
+            k_data = [fast_ma, slow_ma, season_ma]
+            k_pen_colors = ['r', 'y', 'w']
+            fast_mav = triple_golden_cross.calc_fast_mav(volumes)
+            slow_mav = triple_golden_cross.calc_slow_mav(volumes)
+            vol_data = [fast_mav, slow_mav]
+            vol_pen_colors = ['r', 'y']
         elif self.current_indicator_name == self.turtle_info.name:
             turtle = Turtle()
             ups = turtle.calc_batch_up(highs)
