diff --git a/pages/backtest.py b/pages/backtest.py
index 0a2f506..6239123 100644
--- a/pages/backtest.py
+++ b/pages/backtest.py
@@ -10,6 +10,8 @@ from conf.conf import fav_stocks_config_path, FIRST_DAY_YEAR, \
     FIRST_DAY_MONTH, FIRST_DAY_DAY
 from db.models import AStockIndex
 from strategies.boll import BOLL, BOLLBacktest, BOLLInfo
+from strategies.bottom_break_up import BottomBreakUp, BottomBreakUpBacktest, \
+    BottomBreakUpInfo
 from strategies.kdj import KDJ, KDJBacktest, KDJInfo
 from strategies.lucky_duck_head import LuckyDuckHead, LuckyDuckHeadBacktest, \
     LuckyDuckHeadInfo
@@ -234,6 +236,7 @@ class Backtest(QWidget):
         tax = float(self.tax_input.text()) / 1000
         # NEW STRATEGIES #
         boll_info = BOLLInfo()
+        bottom_break_up_info = BottomBreakUpInfo()
         kdj_info = KDJInfo()
         lucky_duck_head_info = LuckyDuckHeadInfo()
         macd_info = MACDInfo()
@@ -245,6 +248,10 @@ class Backtest(QWidget):
             self.backtest_thread = BOLLBacktest(stocks, s_date, e_date,
                                                 init_money, fee,
                                                 pass_fee, tax)
+        elif self.current_strategy_name == bottom_break_up_info.name:
+            self.backtest_thread = BottomBreakUpBacktest(stocks, s_date, e_date,
+                                                         init_money, fee,
+                                                         pass_fee, tax)
         elif self.current_strategy_name == kdj_info.name:
             self.backtest_thread = KDJBacktest(stocks, s_date, e_date,
                                                init_money, fee,
@@ -326,6 +333,7 @@ class Backtest(QWidget):
         tax = float(self.tax_input.text()) / 1000
         # NEW STRATEGIES #
         boll_info = BOLLInfo()
+        bottom_break_up_info = BottomBreakUpInfo()
         kdj_info = KDJInfo()
         lucky_duck_head_info = LuckyDuckHeadInfo()
         macd_info = MACDInfo()
@@ -335,6 +343,8 @@ class Backtest(QWidget):
         wr_info = WRInfo()
         if self.current_strategy_name == boll_info.name:
             backtest = BOLL()
+        elif self.current_strategy_name == bottom_break_up_info.name:
+            backtest = BottomBreakUp()
         elif self.current_strategy_name == kdj_info.name:
             backtest = KDJ()
         elif self.current_strategy_name == lucky_duck_head_info.name:
diff --git a/pages/choose.py b/pages/choose.py
index 06ba289..b626217 100644
--- a/pages/choose.py
+++ b/pages/choose.py
@@ -8,6 +8,7 @@ from conf.conf import fav_stocks_config_path, apply_strategies_config_path, \
     stock_pool_config_path
 from db.models import AStockIndex
 from strategies.boll import BOLLChoose, BOLLInfo
+from strategies.bottom_break_up import BottomBreakUpChoose, BottomBreakUpInfo
 from strategies.common import get_stat_date, get_value_info
 from strategies.kdj import KDJChoose, KDJInfo
 from strategies.lucky_duck_head import LuckyDuckHeadChoose, LuckyDuckHeadInfo
@@ -48,6 +49,7 @@ class Choose(Plots):
         # TODO: make strategies plugin
         # NEW STRATEGIES #
         self.boll_info = BOLLInfo()
+        self.bottom_break_up_info = BottomBreakUpInfo()
         self.kdj_info = KDJInfo()
         self.lucky_duck_head_info = LuckyDuckHeadInfo()
         self.macd_info = MACDInfo()
@@ -320,6 +322,9 @@ class Choose(Plots):
                 # NEW STRATEGIES #
                 if self.apply_strategies[0] == self.boll_info.name:
                     self.choose_thread = BOLLChoose(self.stocks_to_be_chosen)
+                elif self.apply_strategies[0] == self.bottom_break_up_info.name:
+                    self.choose_thread = BottomBreakUpChoose(
+                        self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.kdj_info.name:
                     self.choose_thread = KDJChoose(self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.lucky_duck_head_info.name:
diff --git a/pages/nav.py b/pages/nav.py
index 8b29892..323bc1d 100644
--- a/pages/nav.py
+++ b/pages/nav.py
@@ -12,6 +12,7 @@ from pages.config import Config
 from pages.pool import Pool
 from pages.watch import Watch
 from strategies.boll import BOLLConfig, BOLLInfo
+from strategies.bottom_break_up import BottomBreakUpConfig, BottomBreakUpInfo
 from strategies.kdj import KDJConfig, KDJInfo
 from strategies.lucky_duck_head import LuckyDuckHeadConfig, LuckyDuckHeadInfo
 from strategies.macd import MACDConfig, MACDInfo
@@ -79,6 +80,7 @@ class Nav(QWidget):
         self.percent_change_info = PercentChangeInfo()
         self.lucky_duck_head_info = LuckyDuckHeadInfo()
         self.mcst_info = MCSTInfo()
+        self.bottom_break_up_info = BottomBreakUpInfo()
         self.wr_info = WRInfo()
         self.boll_info = BOLLInfo()
         self.macd_info = MACDInfo()
@@ -120,6 +122,11 @@ class Nav(QWidget):
                 'choose': 'Y' if self.mcst_info.choose_flag else 'N',
                 'watch': 'Y' if self.mcst_info.watch_flag else 'N'
             },
+            {
+                'name': self.bottom_break_up_info.name,
+                'choose': 'Y' if self.bottom_break_up_info.choose_flag else 'N',
+                'watch': 'Y' if self.bottom_break_up_info.watch_flag else 'N'
+            },
             {
                 'name': self.wr_info.name,
                 'choose': 'Y' if self.wr_info.choose_flag else 'N',
@@ -357,6 +364,10 @@ class Nav(QWidget):
             cfg_dlg = LuckyDuckHeadConfig(self)
             cfg_dlg.show()
             cfg_dlg.exec_()
+        if name == self.bottom_break_up_info.name:
+            cfg_dlg = BottomBreakUpConfig(self)
+            cfg_dlg.show()
+            cfg_dlg.exec_()
         if name == self.wr_info.name:
             cfg_dlg = WRConfig(self)
             cfg_dlg.show()
diff --git a/widgets/plots.py b/widgets/plots.py
index 758b852..dcaa455 100644
--- a/widgets/plots.py
+++ b/widgets/plots.py
@@ -5,6 +5,7 @@ from qtpy.QtWidgets import *
 
 from strategies.common import get_latest_batch_data
 from strategies.boll import BOLL
+from strategies.bottom_break_up import BottomBreakUp
 from strategies.kdj import KDJ
 from strategies.lucky_duck_head import LuckyDuckHead
 from strategies.macd import MACD
@@ -462,6 +463,19 @@ class Plots(QWidget):
             k_pen_colors = ['r', 'w', 'y']
             vol_data = []
             vol_pen_colors = []
+        elif self.current_indicator_name == self.bottom_break_up_info.name:
+            bottom_break_up = BottomBreakUp()
+            ma_5 = bottom_break_up.calc_5_ma(closes)
+            ma_10 = bottom_break_up.calc_10_ma(closes)
+            ma_20 = bottom_break_up.calc_20_ma(closes)
+            ma_30 = bottom_break_up.calc_30_ma(closes)
+            ma_60 = bottom_break_up.calc_60_ma(closes)
+            k_data = [ma_5, ma_10, ma_20, ma_30, ma_60]
+            k_pen_colors = ['r', 'y', 'w', 'b', 'g']
+            mav_5 = bottom_break_up.calc_mav(volumes, 5)
+            mav_10 = bottom_break_up.calc_mav(volumes, 10)
+            vol_data = [mav_5, mav_10]
+            vol_pen_colors = ['r', 'y']
         elif self.current_indicator_name == self.lucky_duck_head_info.name:
             lucky_duck_head = LuckyDuckHead()
             fast_ma = lucky_duck_head.calc_fast_ma(closes)
