diff --git a/pages/backtest.py b/pages/backtest.py
index 58fc0da..7b22a90 100644
--- a/pages/backtest.py
+++ b/pages/backtest.py
@@ -12,6 +12,7 @@ from db.models import AStockIndex
 from strategies.boll import BOLL, BOLLBacktest, BOLLInfo
 from strategies.dual_line import DualLine, DualLineBacktest, DualLineInfo
 from strategies.kdj import KDJ, KDJBacktest, KDJInfo
+from strategies.kitten import Kitten, KittenBacktest, KittenInfo
 from strategies.macd import MACD, MACDBacktest, MACDInfo
 from strategies.rsi import RSI, RSIBacktest, RSIInfo
 from strategies.stairs import Stairs, StairsBacktest, StairsInfo
@@ -237,6 +238,7 @@ class Backtest(QWidget):
         boll_info = BOLLInfo()
         dual_line_info = DualLineInfo()
         kdj_info = KDJInfo()
+        kitten_info = KittenInfo()
         macd_info = MACDInfo()
         rsi_info = RSIInfo()
         stairs_info = StairsInfo()
@@ -255,6 +257,10 @@ class Backtest(QWidget):
             self.backtest_thread = KDJBacktest(stocks, s_date, e_date,
                                                init_money, fee,
                                                pass_fee, tax)
+        elif self.current_strategy_name == kitten_info.name:
+            self.backtest_thread = KittenBacktest(stocks, s_date, e_date,
+                                                  init_money, fee,
+                                                  pass_fee, tax)
         elif self.current_strategy_name == macd_info.name:
             self.backtest_thread = MACDBacktest(stocks, s_date, e_date,
                                                 init_money, fee,
@@ -334,6 +340,7 @@ class Backtest(QWidget):
         boll_info = BOLLInfo()
         dual_line_info = DualLineInfo()
         kdj_info = KDJInfo()
+        kitten_info = KittenInfo()
         macd_info = MACDInfo()
         rsi_info = RSIInfo()
         stairs_info = StairsInfo()
@@ -346,6 +353,8 @@ class Backtest(QWidget):
             backtest = DualLine()
         elif self.current_strategy_name == kdj_info.name:
             backtest = KDJ()
+        elif self.current_strategy_name == kitten_info.name:
+            backtest = Kitten()
         elif self.current_strategy_name == macd_info.name:
             backtest = MACD()
         elif self.current_strategy_name == rsi_info.name:
diff --git a/pages/choose.py b/pages/choose.py
index b353abf..1a139d9 100644
--- a/pages/choose.py
+++ b/pages/choose.py
@@ -10,6 +10,7 @@ from db.models import AStockIndex
 from strategies.boll import BOLLChoose, BOLLInfo
 from strategies.dual_line import DualLineChoose, DualLineInfo
 from strategies.kdj import KDJChoose, KDJInfo
+from strategies.kitten import KittenChoose, KittenInfo
 from strategies.macd import MACDChoose, MACDInfo
 from strategies.mcst import MCSTInfo
 from strategies.rsi import RSIChoose, RSIInfo
@@ -46,6 +47,7 @@ class Choose(Plots):
         self.boll_info = BOLLInfo()
         self.dual_line_info = DualLineInfo()
         self.kdj_info = KDJInfo()
+        self.kitten_info = KittenInfo()
         self.macd_info = MACDInfo()
         self.mcst_info = MCSTInfo()
         self.rsi_info = RSIInfo()
@@ -284,6 +286,8 @@ class Choose(Plots):
                         self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.kdj_info.name:
                     self.choose_thread = KDJChoose(self.stocks_to_be_chosen)
+                elif self.apply_strategies[0] == self.kitten_info.name:
+                    self.choose_thread = KittenChoose(self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.macd_info.name:
                     self.choose_thread = MACDChoose(self.stocks_to_be_chosen)
                 elif self.apply_strategies[0] == self.rsi_info.name:
diff --git a/pages/nav.py b/pages/nav.py
index 8e67705..1f81763 100644
--- a/pages/nav.py
+++ b/pages/nav.py
@@ -14,6 +14,7 @@ from pages.watch import Watch
 from strategies.boll import BOLLConfig, BOLLInfo
 from strategies.dual_line import DualLineConfig, DualLineInfo
 from strategies.kdj import KDJConfig, KDJInfo
+from strategies.kitten import KittenConfig, KittenInfo
 from strategies.macd import MACDConfig, MACDInfo
 from strategies.mcst import MCSTInfo
 from strategies.rsi import RSIConfig, RSIInfo
@@ -77,6 +78,7 @@ class Nav(QWidget):
         self.turtle_info = TurtleInfo()
         self.volume_increase_info = VolumeIncreaseInfo()
         self.mcst_info = MCSTInfo()
+        self.kitten_info = KittenInfo()
         self.wr_info = WRInfo()
         self.boll_info = BOLLInfo()
         self.macd_info = MACDInfo()
@@ -113,6 +115,11 @@ class Nav(QWidget):
                 'choose': 'Y' if self.mcst_info.choose_flag else 'N',
                 'watch': 'Y' if self.mcst_info.watch_flag else 'N'
             },
+            {
+                'name': self.kitten_info.name,
+                'choose': 'Y' if self.kitten_info.choose_flag else 'N',
+                'watch': 'Y' if self.kitten_info.watch_flag else 'N'
+            },
             {
                 'name': self.wr_info.name,
                 'choose': 'Y' if self.wr_info.choose_flag else 'N',
@@ -346,6 +353,10 @@ class Nav(QWidget):
             cfg_dlg = VolumeIncreaseConfig(self)
             cfg_dlg.show()
             cfg_dlg.exec_()
+        if name == self.kitten_info.name:
+            cfg_dlg = KittenConfig(self)
+            cfg_dlg.show()
+            cfg_dlg.exec_()
         if name == self.wr_info.name:
             cfg_dlg = WRConfig(self)
             cfg_dlg.show()
diff --git a/widgets/plots.py b/widgets/plots.py
index fa8fa48..10b055d 100644
--- a/widgets/plots.py
+++ b/widgets/plots.py
@@ -7,6 +7,7 @@ from strategies.common import get_latest_batch_data
 from strategies.boll import BOLL
 from strategies.dual_line import DualLine
 from strategies.kdj import KDJ
+from strategies.kitten import Kitten
 from strategies.macd import MACD
 from strategies.mcst import MCST
 from strategies.rsi import RSI
@@ -497,6 +498,13 @@ class Plots(QWidget):
             k_pen_colors = ['y']
             vol_data = []
             vol_pen_colors = []
+        elif self.current_indicator_name == self.kitten_info.name:
+            kitten = Kitten()
+            ma = kitten.calc_batch_ma_indicator(closes)
+            k_data = [ma]
+            k_pen_colors = ['y']
+            vol_data = []
+            vol_pen_colors = []
         else:
             k_data = []
             k_pen_colors = []
